src/pages/Detail/DetailContents/ProductInfo/ProductInfo.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import useFetch from '../../../../hooks/useFetch';
+import BookmarkBorderIcon from '@mui/icons-material/BookmarkBorder';
+import BookmarkIcon from '@mui/icons-material/Bookmark';
+import * as S from './ProductInfo.style';
+import { useNavigate, useParams } from 'react-router-dom';
+import { API } from '../../../../config';
+
+const ProductInfo = ({ info }) => {
+  const params = useParams();
+  const navigate = useNavigate();
+  const [isUpdateWishListFlag, setIsUpdateWishListFlag] = useState(false);
+  let addedWishlist = false;
+
+  const {
+    author_name,
+    description,
+    height,
+    item_name,
+    length,
+    materials,
+    production_year,
+    weight,
+    width,
+  } = info;
+
+  //FIXME - mock data devUrl
+  //const getAllWishList = '/data/wishlistOne.json';
+
+  const getAllWishList = `${API.WISHLIST}`;
+
+  //const interlockToWishList = `${API.WISHLIST}`;
+  const { loading, data } = useFetch(getAllWishList);
+
+  if (!loading) {
+    data.data.map(data => {
+      if (data.item_id === Number(params.id)) {
+        addedWishlist = true;
+        return;
+      }
+    });
+  }
+
+  const addItemToWishList = id => {
+    const addUrl = `${getAllWishList}/items/${id}`;
+    const params = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+    };
+    fetch(addUrl, params).then(res => {
+      if (res.ok) {
+        addedWishlist = !addedWishlist;
+        setIsUpdateWishListFlag(prev => !prev);
+      } else throw new Error(res);
+    });
+    navigate(`/detail/${id}`);
+  };
+
+  const removeItemToWishList = id => {
+    //백엔드 API 준비중
+    /*
+     * 작성될 코드
+     */
```

Comment:
```suggestion
    // TODO : 백엔드 API 준비중
    /*
     * 작성될 코드
     */
```
이런 주석에도 TODO가 붙으면 추적하기 좋겠습니다

Reply:
백엔드쪽에서 wishlist 삭제 로직을 구현하셨는데, 기존에 작성되어있던 wishlist 추가와 동일한 API 주소를 사용한다고 합니다.
따라서 controlWishList 함수 내 삭제, 추가 조건을 별도로 구분짓지 않고
위시리스트 클릭시마다 동일한 API로 추가/삭제 진행되도록 수정완료했습니다.

추가로, 위시리스트 데이터를 연동하고 나서 렌더링이 되지 않는 버그가 있어, useEffect를 통해 같이 수정하였습니다.
참고 부탁드립니다.